프로젝트 세팅

Dae-Hwa opened the pull request:
기본 세팅입니다.

1. git-cz는 일단 포함시키지 않고 가도록 하겠습니다. 좋은 아이디어 있으면 의견 부탁드립니다.

2. 샘플 코드를 따로 작성하고 있는데, 완성본을 이번에 포함 시킬지, 아니면 새로운 풀 리퀘스트로 넘겨서 처리할지 의견 주시면 감사하겠습니다.

3. db관련 세팅은 db 셋업이 완료되면 새롭게 올리도록 하겠습니다.

Dae-Hwa committed:
init spring-boot project version 2.4.2

it contains
- spring-boot-starter-data-jpa
- spring-boot-starter-validation
- spring-boot-starter-hateoas
- spring-boot-configuration-processor
- spring-boot-starter-test
- h2
- postgresql
- lombok
- springdoc-openapi-ui
- springdoc-openapi-data-rest

Dae-Hwa committed:
chore: 환경 별 프로퍼티 및 빌드 스크립트 추가

dev와 prod 리소스 프로퍼티 나누어서 저장 공통은 common으로 저장, 빌드스크립트에 buildProd 추가

SeonHyungJo commented:
@Dae-Hwa DB작업은 추후에 하시는 건가요?

Dae-Hwa commented:
@SeonHyungJo 작성 시점에는 DB 작업을 제외하고 생각했었는데 한꺼번에 세팅하는게 좋을 것 같으면 이어서 커밋하고 다시 요청하는 방향으로 하겠습니다

SeonHyungJo reviewed (approved):
수고하셨습니다. 차차 이후 작업을 진해하시죠

Dae-Hwa commented:
@SeonHyungJo 혹시 git-cz에 대한 의견도 있으시다면, 추가로 해주시면 감사하겠습니다
현재는 npx 를 이용해서 억지로? 사용하고 있는데,
별도의 npm 패키지를 포함시켜 프로젝트를 구성하는게 괜찮은건지 싶긴 하네요

SeonHyungJo commented:
> @SeonHyungJo 혹시 git-cz에 대한 의견도 있으시다면, 추가로 해주시면 감사하겠습니다
> 현재는 npx 를 이용해서 억지로? 사용하고 있는데,
> 별도의 npm 패키지를 포함시켜 프로젝트를 구성하는게 괜찮은건지 싶긴 하네요

억지로 라기 보다는 개인적으로 전역으로 사용하시는건 불편하신가요? 한 번 등록하면 굳이 프로젝트에 추가하지 않아도 되서요
또는 좀 더 찾아보고 있는데 Gradle Plugin이 있긴하더라고요 또는 Git hook을 직접 건드려야겠죠

아니면 직접 커밋 linting관리를 해야해서 안좋다고 생각합니다.
@Dae-Hwa

Dae-Hwa committed:
chore: deprecated 된 gradle 문법 수정

complie -> implementation으로 수정

Dae-Hwa committed:
chore: git-cz용 스크립트 및 배치파일 작성

npx로 git-cz 이모티콘 제거 옵션 추가하여 실행. cmd는 commit, 파워쉘은 ./commit, 배쉬는
./commit.sh 입력하여 실행

Dae-Hwa commented:
git cz는 마땅한 대안을 찾기 힘들어 스크립트로 짜놨습니다.
추후에 README 작성되면 해당 내용 추가하겠습니다.

BKJang commented on build.gradle:
```diff
@@ -0,0 +1,71 @@
+    /**
+     * Database dependencies
+     */
+    runtimeOnly 'com.h2database:h2' // It would be used until the database setup is done.
```
h2 db는 postgre 연결되면 없앨 예정이시죠?

Dae-Hwa replied:
계속 가져갈지 고민 중입니다. 회의 때 의견 잠깐 나눠보시죠

Dae-Hwa replied:
우선 DB 세팅이 늦어질 경우를 대비했습니다. 이 부분은 프런트에서 해당 부분에 상관없이 개발진행이 가능하다는 답변 받았습니다.

가장 큰 장점은 항상 똑같은 상태로 초기화시켜 시작할 수 있다는 것이라 생각했습니다. 그리고 로컬에 각자 DB세팅을 하지 않아도 개발DB로 활용할 수 있으니 여러 사람이 동시에 하나의 스키마를 사용하다가 충돌(예상치 못한 생성 삭제 수정 등)이 일어날 경우도 대비할 수 있지 않을까 생각했습니다.
만약 해당 부분이 크게 문제되지 않을 것 같다면 개발 환경에서도 postgres를 연결시켜도 무방할 것 같습니다. 

하지만 그럼에도 위에서 말씀드린 문제로 단위테스트에 애로사항이 발생할 수 있기 때문에 개발이나 빌드시 단위테스트용으로 설정을 변경해서 가져가야 할 것 같습니다.

개인적으로 개발에는 인메모리를 사용하는게 편리할 것 같은데, 그렇다면 테스트 db와 연결된 프로파일을 하나 추가하는 것도 한 방법일 것 같습니다.

BKJang replied:
이해했습니다. 백엔드 개발 상 인메모리 디비를 쓰는게 편하다면 그 방향으로 가는게 맞을 것 같습니다. 답변 감사합니다 🙏

BKJang reviewed (approved):
샘플 코드는 다음 PR로 넘겨서 작업하시면 좋을 것 같습니다.
질문만 답해주세영 고생하셨습니다 👍

voidmatcha reviewed (approved):
봉기님과 논의중인 부분만 명확해지면 될 것 같네요. 

수고하셨습니다!
